lib/Cake/Routing/Router.php

```diff
@@ -619,8 +619,12 @@ public static function parse($url) {
 			$out['action'] = $out['prefix'] . '_' . $out['action'];
 		}
 
-		if (!empty($ext) && !isset($out['ext'])) {
-			$out['ext'] = $ext;
+		if (!empty($ext)) {
+			if (!isset($out['ext'])) {
+				$out['ext'] = $ext;
+			} elseif (is_array($out['ext'])) {
+				$out['ext'] = $ext;
```

Comment:
These two conditions do the same thing, so the branches can probably be collapsed.